### PR TITLE
WebAssembly | Memory - add note about allocation failure

### PR DESCRIPTION
**Pull request by ghost**
### Description

Allocating `WebAssembly.Memory` is allowed to sporadically fail should the UA decide to do so or be unable to do so, as per the specification.

### Motivation

It conveys the error that may occur if the existing described cases do not apply.

### Additional details

From https://webassembly.github.io/spec/js-api/#dom-memory-memory.

### CODE REVIEW

File: files/en-us/webassembly/javascript_interface/memory/memory/index.md [hunk 1]
```diff
@@ -44,6 +44,8 @@ new WebAssembly.Memory(memoryDescriptor)
 - If `initial` is not specified, a {{jsxref("TypeError")}} is thrown.
 - If `maximum` is specified and is smaller than `initial`, a {{jsxref("RangeError")}} is thrown.
 - If `shared` is present and `true`, yet `maximum` is not specified, a {{jsxref("TypeError")}} is thrown.
+- If `initial` exceeds 65,536 (2^16), a {{jsxref("RangeError")}} is thrown. 2^16 * 64KiB = 4GiB, which is the maximum range that a 32-bit integer can address.
+- If allocation fails, a {{jsxref("RangeError")}} is thrown. This may occur due to attempting to allocate too much at once, or if the page is otherwise out of memory.
```
Comment (Phosra) [inline files/en-us/webassembly/javascript_interface/memory/memory/index.md:48]: Is it okay to have these notes here inline? I consider it cluttered. Suggestions?

Comment (wbamberg) [reply files/en-us/webassembly/javascript_interface/memory/memory/index.md]: The standard way to document exceptions is with a `dl` whose `dt` is the exception type, and whose `dd` can list the conditions. See e.g. https://developer.mozilla.org/en-US/docs/Web/API/IDBIndex/count#exceptions and https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Page_types/API_method_subpage_template#exceptions.

Comment (Phosra) [reply files/en-us/webassembly/javascript_interface/memory/memory/index.md]: Oh right, I've written like that before on other pages here, ha. Will implement.

**Review by wbamberg — APPROVED** [review]
👍 yes, this looks a lot better, thank you @Phosra ! Approving, but not merging because this has Prettier errors. Would you be able to fix them? I'm sorry about our rather contributor-hostile CI :(.

**Comment by Phosra** [comment]
> 👍 yes, this looks a lot better, thank you @Phosra ! Approving, but not merging because this has Prettier errors. Would you be able to fix them? I'm sorry about our rather contributor-hostile CI :(.

Unfortunately for myself (I had to mention it elsewhere too https://github.com/mdn/content/pull/27018#issuecomment-1566288879), I don't have a simple means of doing so on my local system. May I have any suggestions from yourself?

**Comment by wbamberg** [comment]
> > 👍 yes, this looks a lot better, thank you @Phosra ! Approving, but not merging because this has Prettier errors. Would you be able to fix them? I'm sorry about our rather contributor-hostile CI :(.
> 
> Unfortunately for myself (I had to mention it elsewhere too [#27018 (comment)](https://github.com/mdn/content/pull/27018#issuecomment-1566288879)), I don't have a simple means of doing so on my local system. May I have any suggestions from yourself?

I'm asking the other maintainers if we can change the way we do this, as it is (to me) a systemic issue. So we'll have to see what happens. Maybe some other helpful person will fix your PR in the meantime (but the systemic issue will remain).

**Comment by Josh-Cena** [comment]
@Phosra @wbamberg The trick is to paste the entire file into https://prettier.io/playground and paste the result back.

**Comment by bsmth** [comment]
Thanks all, merging shortly as everything looks ready to go 👍🏻

**Pull request merged**